From 3ce40776cf1e9cbb02279117298e2514aba42ffe Mon Sep 17 00:00:00 2001
From: Chris Novakovic <chris@chrisn.me.uk>
Date: Sun, 27 May 2018 17:02:53 +0100
Subject: timeserver: Use gateways as timeservers if
 UseGatewaysAsTimeservers=true

Rather than always assuming that service gateways are capable of
functioning as timeservers, only do so if UseGatewaysAsTimeservers is
set to true in main.conf.
---
 src/timeserver.c | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/src/timeserver.c b/src/timeserver.c
index 393c64c..f9467a2 100644
--- a/src/timeserver.c
+++ b/src/timeserver.c
@@ -268,15 +268,20 @@ GSList *__connman_timeserver_get_all(struct connman_service *service)
 	for (i = 0; service_ts && service_ts[i]; i++)
 		list = __connman_timeserver_add_list(list, service_ts[i]);
 
-	network = __connman_service_get_network(service);
-	if (network) {
-		index = connman_network_get_index(network);
-		service_gw = __connman_ipconfig_get_gateway_from_index(index,
-			CONNMAN_IPCONFIG_TYPE_ALL);
-
-		/* Then add Service Gateway to the list */
-		if (service_gw)
-			list = __connman_timeserver_add_list(list, service_gw);
+	/*
+	 * Then add Service Gateway to the list, if UseGatewaysAsTimeservers
+	 * configuration option is set to true.
+	 */
+	if (connman_setting_get_bool("UseGatewaysAsTimeservers")) {
+		network = __connman_service_get_network(service);
+		if (network) {
+			index = connman_network_get_index(network);
+			service_gw = __connman_ipconfig_get_gateway_from_index(index,
+				CONNMAN_IPCONFIG_TYPE_ALL);
+
+			if (service_gw)
+				list = __connman_timeserver_add_list(list, service_gw);
+		}
 	}
 
 	/* Then add Global Timeservers to the list */
-- 
cgit v1.1

